### ISSUE TITLE
dw_client hangs on replies to single requests never coming, when node is multi-threaded

### ISSUE THREAD

**Issue by tomcucinotta** [tomcucinotta/distwalk#54]
quite easy to reproduce, just try a few times this:
```
./dw_node --nt=2 &
./dw_client -C 1 --nt 16 -n 2
```
If node is started with --nt=1, this doesn't seem to happen (at least, not so easily).

**Comment by tomcucinotta** [comment]
Perhaps useful.... if we restart dw_node or each dw_client run, then the problem doesn't manifest. It happens soon if we keep dw_node up for a few subsequent launches of the client, see logs for details.

[node_log.txt](https://github.com/user-attachments/files/17046885/node_log.txt)
[client_log.txt](https://github.com/user-attachments/files/17046887/client_log.txt)

**Comment by tomcucinotta** [comment]
client computed no elapsed for thr_id 14
```
$ for ((i=0; i<16; i++)); do echo $i $(grep -c "thr_id: $i," client_log.txt); done
0 2
1 2
2 2
3 2
4 2
5 2
6 2
7 2
8 2
9 2
10 2
11 2
12 2
13 2
14 0
15 2
```
and recvw-14 is the only one that got 2 messages in a single receive:
```
$ grep RECV client_log.txt 
[13769.475210620][recvw-0] RECV returned: 512
[13769.475233916][recvw-1] RECV returned: 512
[13769.475362187][recvw-3] RECV returned: 512
[13769.475428922][recvw-2] RECV returned: 512
[13769.482212373][recvw-4] RECV returned: 512
[13769.482293057][recvw-4] RECV returned: 512
[13769.483072707][recvw-6] RECV returned: 512
[13769.483148844][recvw-6] RECV returned: 512
[13769.489376567][recvw-3] RECV returned: 512
[13769.489449932][recvw-0] RECV returned: 512
[13769.489570024][recvw-5] RECV returned: 512
[13769.489689200][recvw-5] RECV returned: 512
[13769.490092904][recvw-7] RECV returned: 512
[13769.490133253][recvw-7] RECV returned: 512
[13769.490391675][recvw-2] RECV returned: 512
[13769.490452371][recvw-1] RECV returned: 512
[13769.490539591][recvw-8] RECV returned: 512
[13769.490611517][recvw-8] RECV returned: 512
[13769.490868217][recvw-10] RECV returned: 512
[13769.490922148][recvw-10] RECV returned: 512
[13769.501761739][recvw-9] RECV returned: 512
[13769.501816056][recvw-9] RECV returned: 512
[13769.502483814][recvw-14] RECV returned: 1024
[13769.508633755][recvw-11] RECV returned: 512
[13769.508684492][recvw-11] RECV returned: 512
[13769.509536485][recvw-13] RECV returned: 512
[13769.509578940][recvw-13] RECV returned: 512
[13769.509693007][recvw-15] RECV returned: 512
[13769.509749422][recvw-15] RECV returned: 512
[13769.510208787][recvw-12] RECV returned: 512
[13769.510248296][recvw-12] RECV returned: 512
```

**Comment by tomcucinotta** [comment]
Indeed, the client was mistakenly processing only the first received message, then getting stuck in a further recv() without finishing processing the other messages already received, if more than 1. Fixed in commit 255b1c2, closing.
